### PR TITLE
Add cycles and tracks keywords to address #101

### PR DESCRIPTION
**Pull request by tsutterley**
Added keywords `cycles` and `tracks` to be able to reduce a search query based on 91-day orbital cycle or RGT.  Uses the CMR `orbit_number` parameter to reduce the query.  Updated Jupyter notebook to show functionality and add notes.

### CODE REVIEW

**Comment by tsutterley** [comment]
this did work locally :disappointed:

**Comment by JessicaS11** [comment]
@tsutterley These are great! Are you suggesting that this no longer works though (I think the functions it uses are largely untested so far, though it does pass Travis - maybe we could work on adding a few tests with this PR? I'd be happy to help with that.)? I still need to check out the notebook changes and test the functionality on my local system...

**Comment by tsutterley** [comment]
It didn't pass Travis initially.  I edited one of the tests to add the optional orbit_number to the CMR parameters.  Should be good to go at this point @JessicaS11

**Comment by JessicaS11** [comment]
Sorry - apparently I forgot to press go on this review before!

File: icepyx/core/icesat2data.py [hunk 1]
```diff
@@ -666,8 +745,40 @@ def avail_granules(self, ids=False):
...
+                # TRK: Reference Ground Track (RGT)
+                # CYCL: Orbital Cycle
+                # GRAN: Granule region (1-14)
+                # RL: Data Release
+                # VERS: Product Version
+                # AUX: Auxiliary flags
+                # SFX: Suffix (h5)
+                PRD,HEM,YY,MM,DD,HH,MN,SS,TRK,CYCL,GRAN,RL,VERS,AUX,SFX=rx.findall(i).pop()
+                cycle_list.append(CYCL)
+                track_list.append(TRK)
```
[the remark this answers is not in the document]
Comment (tsutterley) [reply icepyx/core/icesat2data.py]: Great point.  These have been moved into the granules module.

File: icepyx/core/icesat2data.py [hunk 2]
```diff
@@ -632,9 +708,9 @@ def earthdata_login(self, uid, email):
         self._email = email
 
     # DevGoal: check to make sure the see also bits of the docstrings work properly in RTD
-    def avail_granules(self, ids=False):
+    def avail_granules(self, ids=False, cycles=False, tracks=False):
```
[the remark this answers is not in the document]
Comment (tsutterley) [reply icepyx/core/icesat2data.py]: keywords have been added to the docstring

File: icepyx/core/granules.py [hunk 3]
```diff
@@ -43,6 +44,55 @@ def gran_IDs(grans):
...
+        # CYCL: Orbital Cycle
+        # GRAN: Granule region (1-14)
+        # RL: Data Release
+        # VERS: Product Version
+        # AUX: Auxiliary flags
+        # SFX: Suffix (h5)
+        PRD,HEM,YY,MM,DD,HH,MN,SS,TRK,CYCL,GRAN,RL,VERS,AUX,SFX = \
+            rx.findall(gran["producer_granule_id"]).pop()
+        gran_tracks.append(TRK)
+    return gran_tracks
```
Comment (JessicaS11) [inline icepyx/core/granules.py:95]: Best practices question (I have no idea what the correct answer is, if there even is one): it seems like `gran_IDs()`, `gran_cycles()`, and `gran_RGTs()` all do almost the same thing. Should we be thinking about turning them into one method? I'm just thinking that if you're querying enough of the archive you could potentially see performance improvements if you only iterate through the granule list once.

Comment (tsutterley) [reply icepyx/core/granules.py:95]: They do almost exactly the same thing.  A few commits ago they were a single function with kwargs.  I'll put it back that way :+1:

File: icepyx/core/query.py [hunk 4]
```diff
@@ -252,11 +281,50 @@ def end_time(self):
...
+        >>> reg_a = icepyx.query.Query('ATL06',[-55, 68, -48, 71],['2019-02-20','2019-02-28'])
+        >>> reg_a.tracks
+        ['0841', '0849', '0902', '0910']
+        """
+        return sorted(set(self._tracks))
+
+    @property
+    def orbit_number(self):
+        """
+        Return the ICESat-2 CMR orbit number
```
Comment (JessicaS11) [inline icepyx/core/query.py:313]: for the test case where all tracks/cycles are used, no orbit number(s) are returned. Is this the expected behavior?

File: icepyx/core/granules.py (hunk 3 above)
Comment (JessicaS11) [reply icepyx/core/granules.py:95]: Cool - thanks for being patient with my learning curve!

File: icepyx/core/query.py (hunk 4 above)
Comment (tsutterley) [reply icepyx/core/query.py:313]: yes that's the expected behavior as it'd be an unnecessary CMR parameter (basically would run as icepyx is now)

**Pull request merged**